### ISSUE TITLE
No overload matches this call in default parameters

### ISSUE THREAD

**Issue by efreibe** [microsoft/TypeScript#40122]
Issue Type: <b>Bug</b>

I'm getting this error in my javascript file with ```"javascript.implicitProjectConfig.checkJs": true```

This code doesn't generate error:
```
startOf(unitOfTime, interval = 0, unit) {
    return this._now.startOf(unitOfTime).add(interval, unit).add(this._shift, 'hour').format(this._fmt)
}
```

This code works too:
```
startOf(unitOfTime, interval = 0, unit) {
    return this._now.startOf(unitOfTime).add(interval, 'day').add(this._shift, 'hour').format(this._fmt)
}
```

But this code gives me an error:
```
startOf(unitOfTime, interval = 0, unit = 'day') {
    return this._now.startOf(unitOfTime).add(interval, unit).add(this._shift, 'hour').format(this._fmt)
}
```

The error is:
```
No overload matches this call.
  Overload 1 of 2, '(amount?: DurationInputArg1, unit?: DurationConstructor): Moment', gave the following error.
    Argument of type 'string' is not assignable to parameter of type 'DurationConstructor'.
  Overload 2 of 2, '(unit: DurationConstructor, amount: string | number): Moment', gave the following error.
    Argument of type 'number' is not assignable to parameter of type 'DurationConstructor'.
```

VS Code version: Code 1.48.0 (db40434f562994116e5b21c24015a2e40b2504e6, 2020-08-13T08:03:27.059Z)
OS version: Darwin x64 19.6.0

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-4770HQ CPU @ 2.20GHz (8 x 2200)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>viz_display_compositor: enabled_on<br>viz_hit_test_surface_layer: disabled_off_ok<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|2, 3, 6|
|Memory (System)|16.00GB (0.29GB free)|
|Process Argv||
|Screen Reader|no|
|VM|25%|
</details>

**Comment by mjbvz** [comment]
This seems to be an issue with the moment typings being overly strict and not accepting generic `string` arguments (only string literal types). 

You can workaround this by annotating the type of the parameter with a JSDoc type:

```js
function startOf3(unitOfTime, interval = 0, /** @type {moment.unitOfTime.DurationConstructor} */ unit = 'day') {
}
```

**Comment by efreibe** [comment]
Excellent! I will do that. Thank you
